kratos/elements/distance_calculation_flux_based_element.h

```diff
@@ -0,0 +1,359 @@
+     * @param rCurrentProcessInfo Current ProcessInfo instance. (input)
+     */
+    void GetDofList(
+        DofsVectorType& rElementalDofList,
+        const ProcessInfo& rCurrentProcessInfo) const override;
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override;
+    
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+    double mCorrectionCoefficient=1.0;
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief Assembles the local system for a laplacian, no coeffs
+     * @return A laplacian systes
+     */
+    // void CalculateLaplacianSystem(
+    //     MatrixType &rLeftHandSideMatrix,
+    //     VectorType &rRightHandSideVector,
+    //     const ProcessInfo &rCurrentProcessInfo);
+
+    /**
+     * @brief Assembles the local system to get a gradient in the direction of the flow
+     * @return A transient diffusion system whose solution can be used to construct a velocity field
+     * this velocity will then be used to compute the flowlenght/filltime in a second step
+     */
+    void CalculatePotentialFlowSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo);
+
+    /**
+     * @brief Assembles the local system to get a the flowlength/filltime
+     * @return A pure convection + sorce term system that solves the flowlength/filltime.
+     * For the FlowLength, the assembled source term is the same as |vel|,
+     * therefore obtaining a gradient if 1 in the direction of the velocity
+     * For the FillTime, the assembled source term is 1.0
+     * therefore the gradient is 1/|vel| in the directiion of the velocity
+     */
+    void CalculateDistanceSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo);
+        
+    //computes characteristic length
+    double ComputeH(BoundedMatrix<double,NumNodes, TDim>& DN_DX);
+
+    void CalculateGaussPointsData(  const GeometryType& rGeometry,
+                                    BoundedVector<double,TNumNodes> &rGaussWeights,
+                                    BoundedMatrix<double,TNumNodes,TNumNodes> &rNContainer,
+                                    array_1d<BoundedMatrix<double, TNumNodes, TDim>, TNumNodes>& rDN_DXContainer 
+                                 );
```

Comment:
```suggestion
    void CalculateGaussPointsData(
        const GeometryType& rGeometry,
        BoundedVector<double,TNumNodes> &rGaussWeights,
        BoundedMatrix<double,TNumNodes,TNumNodes> &rNContainer,
        array_1d<BoundedMatrix<double, TNumNodes, TDim>, TNumNodes>& rDN_DXContainer);
```
Style guide suggestion. Please do so in the other places.

Reply:
Besides doxygen documentation is missing.

Reply:
ok